### PR TITLE
API-8654 v3.3 Adjustments

### PR DESCRIPTION
**Pull request by janowsiany**

### CODE REVIEW

File: configuration/api.go [hunk 1]
```diff
@@ -392,3 +396,61 @@ func (a *API) GetLicenseWebhooksState(opts *ManageWebhooksStateOptions) (*Webhoo
...
+	req.Conditions.Geolocation = geolocation
+	err := a.Call("add_auto_access", &req, &resp)
+
+	return resp.ID, err
+}
+
+// UpdateAutoAccess updates an existing auto access.
+func (a *API) UpdateAutoAccess(id string, groupIDs []int, url, domain *Condition, geolocation *GeolocationCondition, desc, nextID string) error {
+	req := updateAutoAccessRequest{
+		addAutoAccessRequest: addAutoAccessRequest{
```
Comment (litwinow) [inline configuration/api.go:434]: This looks weird, I think it'd be better to extract common `autoAccessRequest` struct (without `add`)

File: coverage.out [hunk 2]
```diff
@@ -0,0 +1,276 @@
+mode: set
```
Comment (litwinow) [inline coverage.out:1]: Do we need this file?

File: customer/api.go [hunk 3]
```diff
@@ -401,3 +401,36 @@ func (a *API) RequestEmailVerification(callbackURI string) error {
...
+		GroupID:     groupID,
+		URL:         url,
+		ChannelType: channelType,
+		Test:        isTest,
+	}, &resp)
+	return &resp, err
+}
+
+// GetConfiguration returns the configuration of a given group in a given version.
+func (a *API) GetConfiguration(groupID int, version string) (*StaticConfiguration, error) {
```
Comment (litwinow) [inline customer/api.go:418]: can we drop `Static` from response's type name?

File: customer/api.go [hunk 4]
```diff
@@ -401,3 +401,36 @@ func (a *API) RequestEmailVerification(callbackURI string) error {
 		CallbackURI: callbackURI,
 	}, &emptyResponse{})
 }
+
+// GetDynamicConfiguration returns the dynamic configuration of a given group. It provides data to call Get Configuration and Get Localization.
+func (a *API) GetDynamicConfiguration(groupID int, url, channelType string, isTest bool) (*DynamicConfiguration, error) {
+	var resp DynamicConfiguration
+	err := a.Call("get_dynamic_configuration", &getDynamicConfigurationRequest{
```
Comment (litwinow) [inline customer/api.go:408]: For all config endpoints:
1. `POST` -> `GET`
2. send params in query string, not json body

File: customer/structures.go [hunk 5]
```diff
@@ -36,3 +41,31 @@ type URLInfo struct {
...
+	Type         string `json:"type"`
+	OnlineValue  string `json:"online_value"`
+	OfflineValue string `json:"offline_value"`
+}
+
+type StaticConfiguration struct {
+	Buttons        []ConfigButton               `json:"buttons"`
+	TicketForm     *Form                        `json:"ticket_form,omitempty"`
+	PrechatForm    *Form                        `json:"prechat_form,omitempty"`
+	AllowedDomains []string                     `json:"allowed_domains,omitempty"` // CHECK: why not in docs?
```
Comment (litwinow) [inline customer/structures.go:65]: So, did you check it? ;)

File: webhooks/testdata/user_removed_from_chat.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+	"secret_key": "dummy_key",
+	"action": "user_removed_from_chat",
+	"license_id": 21377312,
+	"payload": {
+		"chat_id": "PJ0MRSHTDG",
+		"thread_id": "K600PKZON8",
+		"user_type": "agent",
+		"user_id": "agent@livechatinc.com",
+    "reason": "manual",
+    "requester_id": "smith@example.com"
```
Comment (litwinow) [inline webhooks/testdata/user_removed_from_chat.json:12]: OPT: fix spacing

**Review by litwinow — CHANGES_REQUESTED** [review]

File: configuration/api.go (hunk 1 above)
Comment (janowsiany) [reply configuration/api.go]: I think i remove weird part, is it better now?

File: coverage.out (hunk 2 above)
Comment (janowsiany) [reply coverage.out]: Removed

File: customer/structures.go (hunk 5 above)
Comment (janowsiany) [reply customer/structures.go]: Created a task for adding allowed_domains to docs

**Comment by litwinow** [comment]
Please also get rid of `/v2` package in the code and go.mod, update readme (/v2) and echo readme

**Comment by grz-gajda** [comment]
@litwinow replaced `v2` to `v3`

**Review by litwinow — APPROVED** [review]

**Pull request merged**